usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -5897,17 +5870,24 @@ static CK_RV rsa_ec_generate_keypair(STDLL_TokData_t * tokdata,
         return CKR_MECHANISM_INVALID;
     }
 
-    if (ktype == CKK_EC)
-        curve_type = get_curve_type_from_attrs(pPublicKeyTemplate,
-                                               ulPublicKeyAttributeCount);
+    if (ktype == CKK_EC) {
+        if (!ep11tok_ec_curve_supported2(tokdata, publ_tmpl, &curve)) {
+            TRACE_ERROR("Curve not supported.\n");
+            return CKR_CURVE_NOT_SUPPORTED;
+        }
+    }
 
-    rc = build_ep11_attrs(tokdata, publ_tmpl, &new_publ_attrs, &new_publ_attrs_len, ktype, CKO_PUBLIC_KEY, curve_type);
+    rc = build_ep11_attrs(tokdata, publ_tmpl,
+                          &new_publ_attrs, &new_publ_attrs_len,
+                          ktype, CKO_PUBLIC_KEY, curve->curve_type);
```

Comment:
what if variable curve is NULL here (e.g. non EC key)? 
Better use something like 
`curve != NULL ? curve->curve_type : -1`